mesh_map/src/mesh_map.cpp

```diff
@@ -151,50 +147,106 @@ MeshMap::MeshMap(tf2_ros::Buffer& tf, const rclcpp::Node::SharedPtr& node)
   vertex_colors_pub = node->create_publisher<mesh_msgs::msg::MeshVertexColorsStamped>("~/vertex_colors", rclcpp::QoS(1).transient_local());
   vector_field_pub = node->create_publisher<visualization_msgs::msg::Marker>("~/vector_field", rclcpp::QoS(1).transient_local());
   config_callback = node->add_on_set_parameters_callback(std::bind(&MeshMap::reconfigureCallback, this, std::placeholders::_1));
+
+  save_service = node->create_service<std_srvs::srv::Empty>("~/save_map", [this](
+    const std::shared_ptr<std_srvs::srv::Empty::Request> request,
+    std::shared_ptr<std_srvs::srv::Empty::Response>      response)
+  {
+    saveLayers();
+  });
 }
 
 bool MeshMap::readMap()
-{
-  RCLCPP_INFO_STREAM(node->get_logger(), "server url: " << srv_url);
-  bool server = false;
-
-  if (!srv_url.empty())
+{ 
+  if(!mesh_io_ptr)
   {
-    server = true;
+    if(!mesh_file.empty() && !mesh_part.empty())
+    {
+      
+      if(mesh_working_file == "")
+      {
+        // default: mesh_working_file = mesh_file filename in this directory
+        mesh_working_file = fs::path(mesh_file).filename().replace_extension(".h5");
+      }
 
-    mesh_io_ptr = std::shared_ptr<lvr2::AttributeMeshIOBase>(
-        new mesh_client::MeshClient(srv_url, srv_username, srv_password, mesh_layer));
-    auto mesh_client_ptr = std::static_pointer_cast<mesh_client::MeshClient>(mesh_io_ptr);
+      if(mesh_working_part == "")
+      {
+        mesh_working_part = mesh_part;
+        RCLCPP_INFO_STREAM(node->get_logger(), "Mesh Working Part is empty. Using mesh part es default: '" << mesh_working_part << "'");
+      } else {
+        RCLCPP_INFO_STREAM(node->get_logger(), "Mesh Working Part is *not* empty: '" << mesh_working_part << "'");
```

Comment:
I feel like the string sounds like the mesh working part not being empty is somewhat out of the ordinary. Maybe change to something like:
```suggestion
        RCLCPP_INFO_STREAM(node->get_logger(), "Using mesh working part from parameter: '" << mesh_working_part << "'");
```